### DISCUSSION TITLE
WN 0003:  Unknown Aggregate and Analytic Functions

### DISCUSSION THREAD

**Discussion by lloydtabb** [malloydata/malloy#1415] [What's Next: Language Proposals]
See https://github.com/malloydata/whatsnext/blob/main/wns/WN-0003/wn-0003.md

**Comment by lloydtabb** [comment]
@mtoy-googly-moogly @carlineng @christopherswenson we hit this twice this week in slack.  I'm not attached to the syntax, but doing something simple here would be great.

**Reply by carlineng** [reply]
I've also heard reports that the inability to run unknown aggregates and window functions is a big gap, and users are needing to kick out to SQL Turduckens very often. One user told me in DMs that this is their biggest pain point with the language at the moment.

**Comment by christopherswenson** [comment]
I am okay accepting this as long as we make it an experimental feature. 

There are two other features which I think are better candidates for solving this problem, but both are harder to implement / probably a little further off:

- [SQL expressions (WN-0004)](https://github.com/malloydata/whatsnext/blob/main/wns/WN-0004/wn-0004.md)
- Custom functions (possibly part of [parameters (WN-0002)](https://github.com/malloydata/whatsnext/blob/main/wns/WN-0002/wn-0002.md))

**Reply by christopherswenson** [reply]
Also I would probably want the experimental syntax to be `some_func!measure<number>(...)`
